warehouse/router/upload.go

```diff
@@ -123,6 +123,7 @@ type UploadJob struct {
 		numStagedEvents                    stats.Measurement
 		uploadSuccess                      stats.Measurement
 		stagingLoadFileEventsCountMismatch stats.Measurement
+		eventDeliveryTime                  stats.Measurement
```

Comment:
I know you copied from above but just using `Measurement` is wrong in this case. Can you use `stats.Timer` which makes more sense?

Reply:
I used `stats.Measurement` because [job.timerStat](https://github.com/rudderlabs/rudder-server/blob/master/warehouse/router/upload_stats.go#L35) returns `stats.Measurement` type. We can change it.

Reply:
That's because `stats.NewTaggedStat` returns a `Measurement`. It has to because it can return anything. The object you're initializing can be more precise though so it makes sense to use the correct interface as the receiving type. For the returning type we can't do much. Sure we can change `job.timerStat` but not `stats.NewTaggedStat` so the underlying problem would be there anyway.

Thanks for making the change :+1: